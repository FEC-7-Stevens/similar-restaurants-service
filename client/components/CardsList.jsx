import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

// eslint-disable-next-line import/extensions
import { CardEntry } from './CardEntry.jsx';

const GridStructure = styled.div`
  width: 100%;
  max-width: 1440px;
  max-height: 774px;
  display: flex;
  flex-wrap: wrap;
`;

const CardsList = ({ similarRestaurants, photos, handleClick }) => {
  return (
    <GridStructure>
      {similarRestaurants.map((restaurant, index) => (
        <CardEntry
          key={restaurant.rid}
          id={restaurant.rid}
          name={restaurant.name}
          price={restaurant.price}
          ratingLabel={restaurant.rating_label}
          ratingScore={restaurant.rating_score}
          category={restaurant.category}
          description={restaurant.description}
          neighborhood={restaurant.neighborhood}
          photos={photos[index]}
          handleClick={handleClick}
        />
      ))}
    </GridStructure>
  );
};

CardsList.propTypes = {
  similarRestaurants: PropTypes.arrayOf(PropTypes.object).isRequired,
  photos: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.object)).isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default CardsList;
