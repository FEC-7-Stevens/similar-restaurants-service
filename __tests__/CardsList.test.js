import React from 'react';
import renderer from 'react-test-renderer';

import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import CardsList from '../client/components/CardsList';
import similarMocks from './../__mocks__/similar.mock.js';
import photosMocks from './../__mocks__/photos.mock.js';

configure({ adapter: new Adapter() });

describe('CardsList', () => {
  const cardsListProps = {
    similarRestaurants: similarMocks.json(),
    photos: photosMocks.json(),
    handleClick: () => {},
  };

  it('renders without crashing', () => {
    shallow(<CardsList {...cardsListProps} />);
  });

  it('renders component correctly', () => {
    const list = mount(<CardsList {...cardsListProps} />);
    expect(list.exists('div')).toEqual(true);
  });

  it('CardsList renders snapshot correctly', () => {
    const component = renderer.create(<CardsList {...cardsListProps} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
