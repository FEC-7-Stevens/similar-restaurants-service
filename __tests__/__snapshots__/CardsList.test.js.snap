// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsList CardsList renders snapshot correctly 1`] = `
<div
  className="CardsList__GridStructure-sc-1og31cb-0 hnKDCc"
>
  <div
    className="CardEntry__Card-sc-16qxg8l-0 ijlrHX"
  >
    <div
      className="PhotosCarousel__Carousel-sc-158497q-0 gfVLiU"
    >
      <img
        alt="restaurant photo"
        className="PhotosCarousel__Slide-sc-158497q-1 koTQkm"
        src="https://zigat.s3-us-west-1.amazonaws.com/image8.jpg"
      />
      <button
        className="PhotosCarousel__Button-sc-158497q-2 cMeYdq btn-left"
        onClick={[Function]}
      >
        &lt;
      </button>
      <button
        className="PhotosCarousel__Button-sc-158497q-2 cMeYdq btn-right"
        onClick={[Function]}
      >
        &gt;
      </button>
    </div>
    <div
      className="CardEntry__CardText-sc-16qxg8l-1 cyGUed textLink"
      onClick={[Function]}
    >
      <div
        className="CardEntry__Name-sc-16qxg8l-2 jllmQB"
      >
        America mint green in
      </div>
      <div
        className="CardEntry__Details-sc-16qxg8l-3 fOzwpo"
      >
        Japanese
        <span>
           • 
        </span>
        Castro
        <span>
           • 
        </span>
        $
      </div>
      <div>
        <img
          alt="logo"
          className="CardEntry__Logo-sc-16qxg8l-4 hokyPJ"
          src="https://zagat.com/assets/img/z-logo-icon-red.svg"
        />
        <span>
           
        </span>
        <span
          className="CardEntry__RatingLabel-sc-16qxg8l-5 cfzkWI"
        >
          Service
        </span>
        <span>
           
        </span>
        <span
          className="CardEntry__RatingScore-sc-16qxg8l-6 iWmVHa"
        >
          4.0
        </span>
      </div>
      <div
        className="CardEntry__Description-sc-16qxg8l-7 fEyeQh"
      >
        Enim non voluptas vel non vel eaque aspernatur.
      </div>
    </div>
  </div>
  <div
    className="CardEntry__Card-sc-16qxg8l-0 ijlrHX"
  >
    <div
      className="PhotosCarousel__Carousel-sc-158497q-0 gfVLiU"
    >
      <img
        alt="restaurant photo"
        className="PhotosCarousel__Slide-sc-158497q-1 koTQkm"
        src="https://zigat.s3-us-west-1.amazonaws.com/image19.jpg"
      />
    </div>
    <div
      className="CardEntry__CardText-sc-16qxg8l-1 cyGUed textLink"
      onClick={[Function]}
    >
      <div
        className="CardEntry__Name-sc-16qxg8l-2 jllmQB"
      >
        et orchid Cary
      </div>
      <div
        className="CardEntry__Details-sc-16qxg8l-3 fOzwpo"
      >
        Japanese
        <span>
           • 
        </span>
        Castro
        <span>
           • 
        </span>
        $$$$
      </div>
      <div>
        <img
          alt="logo"
          className="CardEntry__Logo-sc-16qxg8l-4 hokyPJ"
          src="https://zagat.com/assets/img/z-logo-icon-red.svg"
        />
        <span>
           
        </span>
        <span
          className="CardEntry__RatingLabel-sc-16qxg8l-5 cfzkWI"
        >
          Decor
        </span>
        <span>
           
        </span>
        <span
          className="CardEntry__RatingScore-sc-16qxg8l-6 iWmVHa"
        >
          3.9
        </span>
      </div>
      <div
        className="CardEntry__Description-sc-16qxg8l-7 fEyeQh"
      >
        Error sit vitae doloribus.
      </div>
    </div>
  </div>
</div>
`;
